connector/src/main/scala/com/datastax/spark/connector/rdd/AbstractCassandraJoin.scala

```diff
@@ -108,8 +107,8 @@ private[rdd] trait AbstractCassandraJoin[L, R] {
       .filter(cc => colNames.contains(cc.columnName))
     if (!tableDef.clusteringColumns.startsWith(chosenClusteringColumns)) {
       val maxCol = chosenClusteringColumns.last
-      val maxIndex = maxCol.componentIndex.get
-      val requiredColumns = tableDef.clusteringColumns.takeWhile(_.componentIndex.get <= maxIndex)
+      val maxIndex = tableDef.clusteringColumns.indexOf(maxCol)
+      val requiredColumns = tableDef.clusteringColumns.take(maxIndex + 1)
```

Comment:
Wanted to highlight this for review.  I'm pretty sure the logic I have in there now mirrors what was being done but wanted to make sure this was looked at more closely.